[DCP] Make fsspec public

LucasLLC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121329

Makes  fsspec public

We're going to merge this in https://github.com/pytorch/pytorch/pull/121508 instead due to more CI issues. 

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

LucasLLC committed:
[DCP] Make fsspec public

Makes  fsspec public

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

[ghstack-poisoned]

wz337 commented on torch/distributed/checkpoint/_fsspec_filesystem.py:
```diff
@@ -1,122 +1,15 @@
-    def validate_checkpoint_id(cls, checkpoint_id: Union[str, os.PathLike]) -> bool:
-        return FileSystem.check(checkpoint_id)
+log = logging.getLogger(__name__)
+log.warning(
```
Noice!

wz337 commented:
Discussed offline for public API naming!

LucasLLC commented:
> Discussed offline for public API naming!

Renamed `fsspec_filesystem.py` -> `fsspec.py` as discussed -- thanks for catching this!

LucasLLC committed:
Update on "[DCP] Make fsspec public"


Makes  fsspec public

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

wz337 reviewed (approved):
Thanks for making the changes! LGTM!

fegin reviewed (approved):
Overall, this looks good to me. But I'm not sure if it is necessary to expose the default storage (same question for planner).

LucasLLC commented:
> Overall, this looks good to me. But I'm not sure if it is necessary to expose the default storage (same question for planner).

I'm not sure I'm following, could you point out the line to me?

LucasLLC committed:
Update on "[DCP] Make fsspec public"


Makes  fsspec public

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Make fsspec public"


Makes  fsspec public

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Make fsspec public"


Makes  fsspec public

Differential Revision: [D54596394](https://our.internmc.facebook.com/intern/diff/D54596394/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang

[ghstack-poisoned]
